pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1607,6 +1619,38 @@ func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*ne
 	return nil, errorUtil.New("could not find a default cidr block")
 }
 
+func (n *NetworkProvider) getPodCIDR(ctx context.Context, logger *logrus.Entry) (*net.IPNet, error) {
+	networkConf := &v12.Network{}
+	err := n.Client.Get(ctx, client.ObjectKey{Name: "cluster"}, networkConf)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "failed to get network kind")
+	}
+	var podCIDR *net.IPNet
+	for _, entry := range networkConf.Spec.ClusterNetwork {
+		_, podCIDR, err = net.ParseCIDR(entry.CIDR)
+		if err != nil {
+			panic(err)
+		}
+	}
+	return podCIDR, err
+}
+func (n *NetworkProvider) getServiceCIDR(ctx context.Context, logger *logrus.Entry) (*net.IPNet, error) {
+	networkConf := &v12.Network{}
+	err := n.Client.Get(ctx, client.ObjectKey{Name: "cluster"}, networkConf)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "failed to get network cr")
+	}
```

Comment:
We are fetching this twice for both `getPodCIDR` and `getServiceCIDR`, might be worth fetching one, and passing the object to each function as a parameter.

Reply:
good idea